cca.go

```diff
@@ -0,0 +1,122 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import (
+	"errors"
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix"
+	"github.com/gonum/matrix/mat64"
+)
+
+// CanonicalCorrelations returns the `canonical' correlations between
+// the corresponding pairs of canonical variables -- described by the
+// left and right eigenvectors of the canonical correlation matrix,
+// pvecs and qvecs respectively. The canonical correlation matrix is
+// calculated using the singular value decompositions of the centered
+// inputs, and these singular value decompositions, xsvd and ysvd, are
```

Comment:
Avoid ", and" in these kinds of cases. Two sentences works better; "These singular...".